### PR TITLE
Update Master Model

### PR DESCRIPTION
**Pull request by zhtmike**
- Fixed the scaling value bug in attention cell, and also make the network runnable in pynative mode. Refer to the issue https://gitee.com/mindspore/mindspore/issues/I7M0XX#note_19754663. This modification make the network slightly more accurate.
- Use `inf` value to replace the large value in attention's mask, thus we can remove the type casting of fp32 during the attention calculation safely, it brings us a roughly 5% speed improvement.
- Clean all warning of the dropout in Master model in MindSpore 2.0

Thank you for your contribution to the MindOCR repo.
Before submitting this PR, please make sure:

- [x] You have read the [Contributing Guidelines on pull requests](https://github.com/mindspore-lab/mindcv/blob/main/CONTRIBUTING.md)
- [x] Your code builds clean without any errors or warnings
- [x] You are using approved terminology
- [x] You have added unit tests

## Motivation

(Write your motivation for proposed changes here.)

## Test Plan

(How should this PR be tested? Do you require special setup to run the test or repro the fixed bug?)

## Related Issues and PRs

(Is this PR part of a group of changes? Link the other relevant PRs and Issues here. Use https://help.github.com/en/articles/closing-issues-using-keywords for help on GitHub syntax)

### CODE REVIEW

File: configs/rec/master/master_resnet31.yaml [hunk 1]
```diff
@@ -55,8 +55,8 @@ loss:
 
 scheduler:
   scheduler: warmup_cosine_decay
-  min_lr: 0.00001
-  lr: 0.001
+  min_lr: 0.000005
+  lr: 0.0005
```
Comment (hadipash) [inline configs/rec/master/master_resnet31.yaml:59]: replace these values with `5.0e-6` and `5.0e-4` for easier reading.

**Review by HaoyangLee — APPROVED** [review]

**Comment by HaoyangLee** [comment]
- 此处添加master相关数据
https://github.com/mindspore-lab/mindocr/blob/main/docs/cn/mkdocs/modelzoo_training.md
https://github.com/mindspore-lab/mindocr/blob/main/docs/en/mkdocs/modelzoo_training.md
- 合入后删除旧ckpt和mindir  https://download.mindspore.cn/toolkits/mindocr/master/

File: mindocr/models/utils/attention_cells.py [hunk 2]
```diff
@@ -23,29 +25,27 @@ def __init__(
...
     ) -> Tuple[Tensor, Tensor]:
 
-        d_k = value.shape[-1]
-        score = self.matmul(query, key.transpose(0, 1, 3, 2)) / ms.numpy.sqrt(
-            d_k
-        )  # (N, h, seq_len, seq_len)
-
-        # cast to fp32 to prevent overflow
-        score = ops.cast(score, ms.float32)
+        d_k = float(value.shape[-1])
```
Comment (hadipash) [inline mindocr/models/utils/attention_cells.py:39]: I think it should be `.float()` instead:
```shell
>>> test_var = ms.Tensor([1])
>>> type(test_var)
<class 'mindspore.common.tensor.Tensor'>
>>> type(test_var.float())
<class 'mindspore.common._stub_tensor.StubTensor'>
>>> type(float(test_var))
<class 'float'>
```
`float(Tensor)` gives constant as an output. This means that in the graph mode that constant will be calculated once only during graph construction and not changed during training.

**Comment by hadipash** [comment]
> Use `inf` value to replace the large value in attention's mask, thus we can remove the type casting of fp32 during the attention calculation safely, it brings us a roughly 5% speed improvement.

Don't understand why switching to `np.inf` removes the need for type casting...

File: mindocr/models/utils/attention_cells.py (hunk 2 above)
Comment (zhtmike) [reply mindocr/models/utils/attention_cells.py:39]: it is a integer (which corresponds to a constant value), it is not a tensor

File: configs/rec/master/master_resnet31.yaml (hunk 1 above)
Comment (zhtmike) [reply configs/rec/master/master_resnet31.yaml:59]: yaml does not support scientist format, it will be parsed as string

File: mindocr/models/utils/attention_cells.py (hunk 2 above)
Comment (hadipash) [reply mindocr/models/utils/attention_cells.py:39]: Then no problem!

File: configs/rec/master/master_resnet31.yaml (hunk 1 above)
Comment (hadipash) [reply configs/rec/master/master_resnet31.yaml:59]: It does, you can refer to [here](https://github.com/mindspore-lab/mindocr/blob/main/configs/det/dbnet/db_r50_icdar15.yaml#L56). But the syntax is very strict, for example, `5e-6` will be parsed as string.

**Comment by zhtmike** [comment]
> > Use `inf` value to replace the large value in attention's mask, thus we can remove the type casting of fp32 during the attention calculation safely, it brings us a roughly 5% speed improvement.
> 
> Don't understand why switching to `np.inf` removes the need for type casting...

large number of be out of bound if use fp16 calculation, thus leads to inaccurate result. operator handle `inf` specifically it supports `inf`, e.g., exp(-inf) = 0, needs to a more accurate result in softmax

**Comment by zhtmike** [comment]
> * 此处添加master相关数据
>   https://github.com/mindspore-lab/mindocr/blob/main/docs/cn/mkdocs/modelzoo_training.md
>   https://github.com/mindspore-lab/mindocr/blob/main/docs/en/mkdocs/modelzoo_training.md
> * 合入后删除旧ckpt和mindir  https://download.mindspore.cn/toolkits/mindocr/master/

已更新

File: configs/rec/master/master_resnet31.yaml (hunk 1 above)
Comment (zhtmike) [reply configs/rec/master/master_resnet31.yaml]: updated

**Review by hadipash — APPROVED** [review]
